### ISSUE TITLE
Fix false positives for calculations with parenthesis in selector-descendant-combinator-no-non-space

### ISSUE THREAD

**Issue by wanaryytel** [stylelint/stylelint#3449]
> Clearly describe the bug

I'm getting what I think is a false positive with a line that contains lots of calculations with variables, but doesn't have any descending in it. It's probably related to the variable being in parenthesis, because if you remove the parenthesis then the error is not raised.

> Which rule, if any, is the bug related to?

`selector-descendant-combinator-no-non-space`

> What CSS is needed to reproduce the bug?

```css
div {
    & > :nth-child(#{$i + ($column - 1)}) {
        grid-column: $column;
    }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard"
}

```

> Which version of stylelint are you using?

`9.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`npm run stylelint` and package.json contains `cd static/styles-src && stylelint '**/*.scss'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS calculations.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
> cd static/styles-src && stylelint '**/*.scss'


xx/xx.scss
 13:86  ✖  Unexpected "+($column - 1)"   selector-descendant-combinator-no-non-space
```

**Comment by jeddy3** [comment]
@wanaryytel Thanks for the report and for using the template.

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule). It should just be the case of returning early if the selector is non-standard ([like so](https://github.com/stylelint/stylelint/blob/4d53e92c0830e4dbc106e696c84a6eaeacc02d05/lib/rules/selector-id-pattern/index.js#L39-L41)).
